entrypoint.sh

```diff
@@ -21,16 +21,19 @@ declare -A grades=(
 	['F']=1
 )
 
-if [ "$2" = 'true' ]; then
+if [ "$FOLLOW_REDIRECTS" = 'true' ]; then
     FOLLOW_REDIRECTS='on'
 fi
 
-GRADE=$3
-
 RATING=$(curl -s -L "https://securityheaders.com/?hide=on&followRedirects=$FOLLOW_REDIRECTS&q=$1" -I | sed -En 's/x-grade: (.*)/\1/p' | tr -d '\r')
 
 echo "::set-output name=rating::$RATING"
 
+if [ $RATING = "R" ]; then
```

Comment:
Without this the script fails with a misleading error of `./entrypoint.sh: line 32: [: -ge: unary operator expected` when a redirect is encountered and not followed.

Reply:
Which url are you testing?

Reply:
I was testing with `https://symbolica.dev` which redirects to the canonical version of our site `https://www.symbolica.dev`

Reply:
Ok, I see. I did some testing and this seems to work in all cases.

I think we should change the default to follow redirects, that is the default on the `securityheaders.com` so it should work fine for us also.

We can do it like this if we want 👇🏻 

```bash
FOLLOW_REDIRECTS=${3-'1'}
FOLLOW_REDIRECTS=${FOLLOW_REDIRECTS/off/'0'}
```

If we don't want to change the default we could just do this.

```bash
FOLLOW_REDIRECTS=${3-'0'}
```

Settings `$3` to on enables redirects so in your case you need to specify `GRADE` and `FOLLOW_REDIRECTS` to make it scan your site.

Thoughts?

This is the code for changing the behaviour to follow redirects as default.

```bash
#!/bin/bash

if [ $# == 0 ]; then
    echo "Usage: url [grade] [followRedirects]"
    echo "* url: URL to analyse."
    echo "* grade: The desired security grade of your HTTP response headers. Possible grades: A+, A, B, C, D, E, F. Defaults to B."
    echo "* followRedirects: Follow redirects. Defaults to on, set to off to disable."
    exit 1
fi


GRADE=${2:-'B'}
FOLLOW_REDIRECTS=${3-'1'}
FOLLOW_REDIRECTS=${FOLLOW_REDIRECTS/off/'0'}

declare -A grades=(
	['A+']=7
	['A']=6
	['B']=5
	['C']=4
	['D']=3
	['E']=2
	['F']=1
)

RATING=$(curl -s -L "https://securityheaders.com/?hide=on&followRedirects=$FOLLOW_REDIRECTS&q=$1" -I | sed -En 's/x-grade: (.*)/\1/p' | tr -d '\r')

echo "::set-output name=rating::$RATING"

if [ $RATING = "R" ]; then
    echo "$1 returned a redirect, enable the followRedirects option to scan this URL."
    exit 1
fi

if [ ${grades[$RATING]} -ge ${grades[$GRADE]} ]; then
	exit 0
else
	exit 1
fi
```

Reply:
Yeah this seems like a good solution. I didn't want to change the existing behaviour which is why I defaulted it to off. But I think overall it's better if we default it to on, because I think this is a better default. 

I'll make those changes you've suggested.

Reply:
Out of curiosity what does `FOLLOW_REDIRECTS/off/'0'` do? Set the value to `"off"` if the value is 0?

Reply:
OK, I see it is a regex replace for `off` with `0`.

Reply:
Yes, it just a replacement for off. Sending 0 or '' seems to work as true false for the parameter `followRedirects`. I choose off because the param value for `true` is `on` 😀